libs/liquidity/src/lib/liquidity-data-provider.ts

```diff
@@ -0,0 +1,231 @@
+  ]);
+  const stakeToCcySiska = stakeToCcySiskas && stakeToCcySiskas[0];
+  const liquidityProviders: LiquidityProvision[] | undefined =
+    data?.market?.data?.liquidityProviderFeeShare
+      ?.filter(
+        (p: MarketLiquidity_market_data_liquidityProviderFeeShare) =>
+          !partyId || p.party.id === partyId
+      ) // if partyId is provided, filter out other parties
+      .map(
+        (provider: MarketLiquidity_market_data_liquidityProviderFeeShare) => {
+          const liquidityProvisionConnection =
+            data.market?.liquidityProvisionsConnection.edges?.find(
+              (e) => e?.node.party.id === provider.party.id
+            );
+          const balance =
+            liquidityProvisionConnection?.node?.party.accountsConnection.edges?.reduce(
+              (acc, e) => {
+                return e?.node.type === AccountType.ACCOUNT_TYPE_BOND // just an extra check to make sure we only use bond accounts
+                  ? acc.plus(new BigNumber(e?.node.balance ?? 0))
+                  : acc;
+              },
+              new BigNumber(0)
+            );
+          const obligation =
+            stakeToCcySiska &&
+            new BigNumber(stakeToCcySiska)
+              .times(liquidityProvisionConnection?.node?.commitmentAmount ?? 1)
+              .toString();
+          const supplied =
+            stakeToCcySiska &&
+            new BigNumber(stakeToCcySiska).times(balance ?? 1).toString();
+          return {
+            party: provider.party.id,
+            createdAt: liquidityProvisionConnection?.node?.createdAt,
+            updatedAt: liquidityProvisionConnection?.node?.updatedAt,
+            commitmentAmount:
+              liquidityProvisionConnection?.node?.commitmentAmount,
+            fee: liquidityProvisionConnection?.node?.fee,
+            status: liquidityProvisionConnection?.node?.status,
+            equityLikeShare: provider.equityLikeShare,
+            averageEntryValuation: provider.averageEntryValuation,
+            obligation,
+            supplied,
+          };
+        }
+      );
+  const liquidityData: LiquidityData = {
+    liquidityProviders,
+    suppliedStake: data?.market?.data?.suppliedStake,
+    targetStake: data?.market?.data?.targetStake,
+    decimalPlaces: data?.market?.decimalPlaces,
+    positionDecimalPlaces: data?.market?.positionDecimalPlaces,
+    code: data?.market?.tradableInstrument.instrument.code,
+    assetDecimalPlaces:
+      data?.market?.tradableInstrument.instrument.product.settlementAsset
+        .decimals,
+    symbol:
+      data?.market?.tradableInstrument.instrument.product.settlementAsset
+        .symbol,
+  };
+  return formatLiquidityData(liquidityData);
+};
+
+export const formatLiquidityData = ({
+  suppliedStake,
+  assetDecimalPlaces,
+  targetStake,
+  code,
+  symbol,
+  liquidityProviders,
+}: LiquidityData) => {
+  return {
+    suppliedStake: addDecimalsFormatNumber(suppliedStake, assetDecimalPlaces),
```

Comment:
suggestion (non blocking): Its better to return the raw numbers/bignumbers rather than format them here. What if another component wants to format them in a different way?

Utilize AgGrid valueFormatters instead.

Reply:
updated. before using only ag-grid for this, I had to change the formatting in two places. now I don;t need to do this anymore. moved the value formatters back.